Google.Api.Gax.Grpc/ClientHelper.cs

```diff
@@ -18,15 +19,21 @@ public class ClientHelper
     {
         private readonly CallSettings _clientCallSettings;
         private readonly CallSettings _versionCallSettings;
+        private readonly ILoggerFactory _loggerFactory;
+        private readonly string _baseCategoryName;
 
         /// <summary>
         /// Constructs a helper from the given settings.
         /// Behavior is undefined if settings are changed after construction.
         /// </summary>
         /// <param name="settings">The service settings.</param>
-        public ClientHelper(ServiceSettingsBase settings)
+        /// <param name="loggerFactory">The logger factory to use for API calls</param>
```

Comment:
I'm not sure about the logger per method. It's more usual to have a logger per class. Could we then do some structured logging to add method info etc.?

 I'm ultimately fine either way.

Reply:
Hmm. The trouble is the code that's doing the logging doesn't know which method it's logging. We could potentially put that in ApiCall though.

If we accept an ILogger rather than an ILoggerFactory, that opens the option of clients using different loggers for different use-cases, too. We could either have just an ILogger property on ClientBuilderBase, or have both ILogger and ILoggerFactory and use the factory if we don't have a logger. Let's chat about it.